File: mario-less.c
#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt);
void hash(int n);
void space(int s);
int main(void)
{
    int i = get_positive_int("Enter an integer from 1 to 8: ");
    for (int num = 1; num <= i; num++)
    {
        space(i-num);
        hash(num);
        printf("\n");
    }
}

int get_positive_int(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}

void hash(int n)
{
    for (int j = 0; j < n; j++)
    {
        printf("#");
    }
}

void space(int n)
{
    for (int j = 0; j < n; j++)
    {
        printf(" ");
    }
}

File: Week 6 - Python/mario - less.py

from cs50 import get_int

# main function


def main():
    i = get_positive_int("Height: ")
    pyramid(i)


# check if user input is positive integer

def get_positive_int(prompt):
    while True:
        n = get_int(prompt)
        if n > 0 and n < 9:
            break
    return n


# print number the pyramid

def pyramid(n):
    for i in range(1, n+1):
        print(" "*(n-i), end="")
        print("#"*i)


if __name__ == '__main__':
    main()